pax-web-extender-war/src/main/java/org/ops4j/pax/web/extender/war/internal/UnregisterWebAppVisitorWC.java

```diff
@@ -120,18 +120,15 @@ public void visit(final WebApp webApp) {
 	 */
 	public void visit(final WebAppServlet webAppServlet) {
 		NullArgumentException.validateNotNull(webAppServlet, "Web app servlet");
-		final Class<? extends Servlet> servletClass = webAppServlet
-				.getServletClass();
-		if (servletClass != null) {
-			//CHECKSTYLE:OFF
-			try {
-				webContainer.unregisterServlets(servletClass);
-				webAppServlet.setServletClass(null);
-			} catch (Exception ignore) { 
-				LOG.warn("Unregistration exception. Skipping.", ignore);
-			}
-			//CHECKSTYLE:ON
-		}
+                //Servlet name shouldn't be null
+                final String servletName = webAppServlet.getServletName();
+                //CHECKSTYLE:OFF
+                try {
+                        webContainer.unregisterServlet(servletName);
+                } catch (Exception ignore) { // CHECKSTYLE:SKIP
+                        LOG.error("Unregistration exception. Skipping.", ignore);
+                }
+                //CHECKSTYLE:ON
 	}
 
 	/**

```

Comment:
This when servlets are registered in web.xml. I think removing setServletClass could be to brave, as it could be set outside of OPS. What in situation when there will be few servlets with same class having different names?